unit-test/background/atb.js

```diff
@@ -287,3 +287,63 @@ describe('complex install workflow cases', () => {
             })
     })
 })
+
+describe('atb.getUninstallURL()', () => {
+    it('should update the uninstall URL correctly as the ATB values are updated', async () => {
+        settingHelper.stub({ atb: null })
+
+        let uninstallUrl = null
+        const setUninstallURLSpy = spyOn(browser.runtime, 'setUninstallURL').and.callFake(url => {
+            uninstallUrl = url
+        })
+
+        const checkUninstallUrl = (expectedAtb, expectedSetAtb) => {
+            const { origin, pathname, searchParams } = new URL(uninstallUrl)
+
+            expect(origin).toEqual('https://duckduckgo.com')
+            expect(pathname).toEqual('/atb.js')
+            expect(searchParams.get('uninstall')).toEqual('1')
+            expect(searchParams.get('action')).toEqual('survey')
+            expect(searchParams.get('atb')).toEqual(expectedAtb)
+            expect(searchParams.get('set_atb')).toEqual(expectedSetAtb)
+            expect(searchParams.get('browser')).toEqual('Chrome')
+            expect(searchParams.get('bv')).toEqual('108')
+            expect(searchParams.get('v')).toEqual('1234.56')
+        }
+
+        // Initial state.
+        expect(settings.getSetting('atb')).toEqual(null)
+        expect(setUninstallURLSpy).not.toHaveBeenCalled()
+        expect(uninstallUrl).toEqual(null)
+
+        // Before the atb is set.
+        uninstallUrl = await atb.getSurveyURL()
+        checkUninstallUrl(null, null)
+
+        // After the atb is set.
+        stubLoadJSON({ returnedAtb: 'v119-8' })
+        await atb.setInitialVersions()
+        await new Promise(resolve => setTimeout(resolve, 0))
```

Comment:
This could race, would listening for the even be cleaner?

Reply:
I thought about that too. I think this is OK, since for the unit tests the event is firing synchronously. I'm going to leave it alone for now considering we're in a rush here, but if the test starts failing we can always adjust it later.